### ISSUE TITLE
Colon rule spacing in function/method calls

### ISSUE THREAD

**Issue by kmikael** [realm/SwiftLint#1389]
The colon rule doesn't find correct spacing in function calls for me, for example:

    object.method(x:5, y:"string")

should be

    object.method(x: 5, y: "string")

Is this by design? I couldn't find a configuration to make this work.

**Comment by marcelofabri** [comment]
I think this is not currently supported, based on the [tests](https://github.com/realm/SwiftLint/blob/master/Source/SwiftLintFramework/Rules/ColonRule.swift#L23).

Let's keep it open to track as an enhancement.

**Comment by kmikael** [comment]
There is this case as well that does not work yet:

- `[class|struct|enum|extension] X: Y {`

I think there are a few more but I can't remember them now. I will post them if I do.

**Comment by keith** [comment]
FWIW I think this is probably about the same case as this not erroring:

```swift
func foo(bar: [String : String]) {}
```

**Comment by serges147** [comment]
I'm not sure is this cause but in 0.23.1 "warning: Colon Violation" rule is broken. I've got hundreds of such violations in 0.23.1, but zero at 0.22.0

**Comment by jpsim** [comment]
Please file a new issue, filling out the issue template. It’s hard to help without more information.
